### ISSUE TITLE
First example failing

### ISSUE THREAD

**Issue by andreaferretti** [stefantalpalaru/golib-nim#2]
I am trying to copy [this go example](http://jnwhiteh.net/posts/2010/09/go-examples-1-channels-and-goroutines.html).

Unfortunately, `nimble c hellogo` slains the compiler :-(

Notice that I am still trying to install `golib` (I am opening a separate issue on this on your other repository), but it seems that the compiler crashes long before trying to link anything.

My attempt goes like this:

``` nim
# hellogo.nim
import golib

proc printNumbers(numbers: chan[int], done: chan[bool]) {.goroutine.} =
  var n: int
  for _ in 1 .. 10:
    n = <- numbers
    echo n
  done <- true

proc numberGen(start, count: int, ch: chan[int]) {.goroutine.} =
  for i in 0 .. < count:
    ch <- start + i
  ch.close()

proc goMain() =
  var
    numberChan = makeChan(int)
    done = makeChan(bool)
  go numberGen(1, 10, numberChan)
  go printNumbers(numberChan, done)
  var _  = <- done

when isMainModule:
  golibMain()
```

`hellogo.nimble`:

```
[Package]
name          = "hello go"
version       = "0.1.0"
author        = "Andrea Ferretti"
description   = "hello go"
license       = "Apache2"

bin           = "hellogo"

[Deps]
Requires: "nim >= 0.12.0, golib >= 0.0.1"
```

`nim.cfg`:

```
--threads:on
--stackTrace:off
--passC:"--std=gnu99 -fsplit-stack"
--dynlibOverride:"go"
--passL:"-lgolib -lgo"
--gc:go
```

**Comment by andreaferretti** [comment]
Sorry, my bad, I was missing the `{.gomain.}` pragma
